context.go

```diff
@@ -27,23 +25,30 @@ var (
 	ctxEntriesKey = &ctxEntriesType{}
 )
 
+// WithMap adds the provided Map to the context and returns a new Context.
 func WithMap(ctx context.Context, m Map) context.Context {
 	return context.WithValue(ctx, ctxEntriesKey, m)
 }
 
-func NewContext(ctx context.Context, keyvalues ...core.KeyValue) context.Context {
+// NewContext with one or more KeyValues.
+func NewContext(ctx context.Context, keyvalues ...KeyValue) context.Context {
 	return WithMap(ctx, FromContext(ctx).Apply(MapUpdate{
 		MultiKV: keyvalues,
 	}))
 }
 
+// FromContext extracts a Map from the Context if it exists, otherwise returns
+// an empty Map.
 func FromContext(ctx context.Context) Map {
```

Comment:
I think `otel.FromContext` doesn't capture well what this does. What are we trying to get from context?